test/stateChangeCapture.test.ts

```diff
@@ -1,69 +1,146 @@
+        expect(dynamoPutSpy).toBeCalledWith(expected);
+      })
+
+      it('should set value to 1 when alarm state is ALARM', async () => {
+        await givenNoPreviousStateInDynamo("ALARM", "OK");
+        var expected = { "Item": { "id": "ALARM_flaky-service", "resourceId": "2019-12-12T06:25:41.200+0000", "pipelineName": "flaky-service-pipeline", "bookmarked": "N", "state": "ALARM", "value": 1 }, "TableName": "MetricsEventStore" };
+        expect(dynamoPutSpy).toBeCalledWith(expected);
+      })
+
+      it('should make api call to retrieve the pipeline name ', async () => {
+        await givenNoPreviousStateInDynamo("ALARM", "OK");
+        expect(codePipelineSpy).toBeCalled();
+      })
+    });
+
+    describe("When state is the same", () => {
+      it('should not make any api calls when current and previous states are both ALARM', async () => {
+        await givenNoPreviousStateInDynamo("ALARM", "ALARM");
+        expect(codePipelineSpy).not.toBeCalled();
+        expect(dynamoPutSpy).not.toBeCalled();
+      })
+      it('should not make any api calls when current and previous states are both OK', async () => {
+        await givenNoPreviousStateInDynamo("OK", "OK");
+        expect(codePipelineSpy).not.toBeCalled();
+        expect(dynamoPutSpy).not.toBeCalled();
+      })
+    })
+
+    it('should ignore the state when it is insufficient data', async () => {
+      await givenNoPreviousStateInDynamo("INSUFFICIENT_DATA", "OK");
+      expect(codePipelineSpy).not.toBeCalled();
+      expect(dynamoPutSpy).not.toBeCalled();
+    })
+  })
 
-describe('stateChangeCapture', () => {
-  let dynamoPutSpy;
-  let eventStoreSpy;
-  let alarmHistory: AlarmHistoryItems;
 
-  beforeEach(async () => {
+  async function givenNoPreviousStateInDynamo(currentState: string, prevState: string) {
+    let alarmStateEvent: CloudwatchStateChangeEvent = { ...mockCloudwatchEvent };
+    mockReturnEmptyItemFromDynamo();
+    let items = [
+      {
+        date: '2019-01-02T00:02:30.000Z',
+        state: prevState,
+        oldSate: 'INSUFFICIENT_DATA'
+      },
+      {
+        date: '2019-01-03T00:02:30.000Z',
+        state: 'INSUFFICIENT_DATA',
+        oldSate: prevState
+      },
+      {
+        date: '2019-01-04T00:02:30.000Z',
+        state: currentState,
+        oldSate: 'INSUFFICIENT_DATA'
+      }
+    ]
+    mockCloudwatchHistory(items);
+    alarmStateEvent.detail.state.value = currentState;
+    await handler(alarmStateEvent);
+  }
+
+  async function givenPreviousStateExistsInDynamo(currentState: string, prevState: string) {
+    let alarmStateEvent: CloudwatchStateChangeEvent = { ...mockCloudwatchEvent };
+    mockGetLastItemFromDynamo(prevState);
+    alarmStateEvent.detail.state.value = currentState;
+    await handler(alarmStateEvent);
```

Comment:
same as above.

"don't make me think"

Similar to the idea getters should not have side effects.